ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -133,36 +125,55 @@ export class SubmitWorkflowPanel extends React.Component<Props, State> {
                     title: value
                 }))}
                 onChange={event => {
-                    this.setState({
-                        parameters: this.state.parameters.map(p => ({
-                            name: p.name,
-                            value: p.name === parameter.name ? event.value : this.getValue(p),
-                            enum: p.enum
-                        }))
-                    });
+                    const update = {} as State;
+                    update[parameterStateName] = this.state[parameterStateName].map(p => ({
+                        name: p.name,
+                        value: p.name === parameter.name ? event.value : this.getValue(p),
+                        enum: p.enum
+                    }));
+                    this.setState(update);
```

Comment:
do you want to use `this.setState(oldState => newState)` version here?

Reply:
There is only one example of the pattern you suggested in the code base:
https://github.com/argoproj/argo-workflows/blob/v3.3.1/ui/src/app/shared/components/input-filter.tsx#L63

What would this code gain by using that pattern? All other `setState`s in the UI look like this.

Reply:
it is useful it you only want to modify part of the state, which is what I think you are trying to do here

Reply:
I think I'd do this differently, I'd add a new parameters to `displaySelectFieldForEnumValues` named `onChange` and pass in the correct func to it. The decouples the two functions.

Reply:
>it is useful it you only want to modify part of the state

I'm not convinced that `setState(update)` does _not_ do a partial update. At least, I haven't been able to find any documentation that reads otherwise. I also tried using the arrow function update anyway, but was unable to make it successfully update the state.

>I'd add a new parameters to displaySelectFieldForEnumValues named onChange and pass in the correct func to it

I understand the sentiment here, but I gave up after a few hours of trying to make it work this way. I ran into two problems:
* The update function would lose reference to `this`.
* The `setState` update would appear to work, but didn't actually update any values.

If you have any more specific direction, or just want someone else to make the changes, that would be fine with me.